src/bvals/cc/bvals_cc_in_one.cpp

```diff
@@ -474,11 +560,26 @@ TaskStatus SendBoundaryBuffers(std::shared_ptr<MeshData<Real>> &md) {
 
               Kokkos::parallel_for(
                   Kokkos::ThreadVectorRange(team_member, si, ei + 1), [&](const int i) {
+                    const Real val =
+                        src_allocated ? boundary_info(b).var(v, k, j, i) : 0.0;
                     boundary_info(b).buf(i - si +
-                                         Ni * (j - sj + Nj * (k - sk + Nk * v))) =
-                        boundary_info(b).var(v, k, j, i);
+                                         Ni * (j - sj + Nj * (k - sk + Nk * v))) = val;
+#ifdef ENABLE_SPARSE
+                    if (std::abs(val) > threshold) {
```

Comment:
Ping @Yurlungur and @dholladay00 as you reviewed before:

We should also check for `src_allocated` here, don't we?
Otherwise, a dense field that happens to have `0` in all ghost zones, would not be sent or does this not matter because data will always be sent and just not be processes because the the last index in the buffer will contain a 0?
In the latter case, I may be concerned that the receiving buffer is not "clean". IIRC we never reset the receiving buffers because the original (Athena++) designed assumed that all values will always be communicated (with fresh data).

Reply:
A dense field may have relevant data even if it's below the threshold in magnitude. This is definitely a bug. Good catch.